index.html

```diff
@@ -3476,110 +3497,147 @@ <h2>Processing a Manifest</h2>
+											href="https://infra.spec.whatwg.org/#list-item">items</a>, respectively as
+										defined by <a href="#processing-checks-general">this iterator</a>.</p>
+								</li>
+							</ol>
 						</li>
 
 						<li id="processing-checks-type">
-							<p>(<a href="#publication-types"></a>) If <var>processed["type"]</var> is not set or
-								contains an empty value, this is a <a>validation error</a>. Set its value to an array
-								with the value "<code>CreativeWork</code>".</p>
+							<p>(<a href="#publication-types"></a>) If <var>processed["type"]</var> is not set or is an
+								empty <a href="https://infra.spec.whatwg.org/#list">list</a>, <a>validation error</a>,
+								set to <code>«&#160;"CreativeWork"&#160;»</code>.</p>
 						</li>
 
 						<li id="processing-checks-abridged">
-							<p>(<a href="#abridged"></a>) If <var>processed["abridged"]</var> is set and it is not a <a
-									href="#value-boolean">boolean value</a>, this is a <a>validation error</a>.</p>
+							<p>(<a href="#abridged"></a>) If <var>processed["abridged"]</var> is set and is not a <a
+									href="https://infra.spec.whatwg.org/#boolean">boolean</a>, <a>validation error</a>,
+									<a href="https://infra.spec.whatwg.org/#map-remove">remove</a>
+								<var>processed["abridged"]</var>.</p>
 						</li>
 
 						<li id="processing-checks-duration">
-							<p>(<a href="#duration"></a>) If <var>processed["duration"]</var> is set and its value is
-								not a valid duration value, per&#160;[[!iso8601]], this is a <a>validation
-								error</a>.</p>
+							<p>(<a href="#duration"></a>) If <var>processed["duration"]</var> is set and is not a valid
+								duration value, per&#160;[[!iso8601]], <a>validation error</a>, <a
+									href="https://infra.spec.whatwg.org/#map-remove">remove</a>
+								<var>processed["duration"]</var>.</p>
 						</li>
 
 						<li id="processing-checks-last-mod">
 							<p>(<a href="#last-modification-date"></a>) If <var>processed["dateModified"]</var> is set
-								and its value is not a valid date or date-time per&#160;[[!iso8601]], this is a
-									<a>validation error</a>.</p>
+								and is not a valid date or date-time per&#160;[[!iso8601]], <a>validation error</a>, <a
+									href="https://infra.spec.whatwg.org/#map-remove">remove</a>
+								<var>processed["dateModified"]</var>.</p>
 						</li>
 
 						<li id="processing-checks-pub-date">
-							<p>(<a href="#publication-date"></a>) If <var>processed["datePublished"]</var> is set and
-								its value is not a valid date or date-time per&#160;[[!iso8601]], this is a
-									<a>validation error</a>.</p>
+							<p>(<a href="#publication-date"></a>) If <var>processed["datePublished"]</var> is set and is
+								not a valid date or date-time per&#160;[[!iso8601]], <a>validation error</a>, <a
+									href="https://infra.spec.whatwg.org/#map-remove">remove</a>
+								<var>processed["datePublished"]</var>.</p>
```

Comment:
_Note_: this comment shouldn't block this PR, it can be further discussed in #104 and —if it results in any change— dealt-with in another PR.

do we want to allow smart UA to recover from these? Some might be able to fix an invalid date string, or convert a date in natural language to the ISO format.

One possibility could be to define a generic step like "convert input to the expected type" as a UA-defined step, and either return that or remove.

I'm not sure; what do you think?

Reply:
Specifically for dates this may make sense (I am a little bit afraid we create a precedence, but maybe we are fine), but the step you propose can be read as if the UA _MUST_ try to recover. This may be a step too far (let alone the fact that we would have to test this, and I am not sure how many different strings there are that can be interpreted correctly). 

"... is set and is not a valid date or date-time per [[!iso8601]], validation error. If the User Agent is able to interpret the value as a valid date or date-time, replace the value with that generated value; otherwise remove it."

Or something like that, converted to infra-speak...

Reply:
Doesn't this lead to interop problems if we can't define how to fix the invalid input? There are a whole host of reasons why the input could be invalid, from the data type being wrong to the date string not conforming.

Reply:
Yeah... what is worse, it could lead to wrong interpretations. E.g., what date is meant by "01/02/03"? 1st of February 2003 (in the UK), 2nd of January 2001 (in the US) and, I believe, 3rd of February in 2001 in Japan (or in Hungary, b.t.w.).

Maybe we should keep it as is after all.

Reply:
> but the step you propose can be read as if the UA _MUST_ try to recover.

sorry I wasn't clear. I was suggesting a MAY at best 🙂

> Doesn't this lead to interop problems if we can't define how to fix the invalid input? 

yeah, that's why I'm not sure.It's not so uncommon to have implementation-defined behavior/processing. There's not a huge risk for interoperability I think, given the nature of the metadata. At worst there will be cases like users not getting their publications sorted in the same order on various UAs if they sort it by date, maybe this isn't too bad? Again, these are open questions, I don't have a strong opinion on this.

> it could lead to wrong interpretations

Right. I think that given a language and a date string, a lot can be done (e.g. "Dec 2019", "01/02/03" are not unambiguous in US English). I'm more worried of implementation starting using a default date instead of removing the value if they can't make sense of the authored date.

As always it would be interesting to see if "invalid" dates are commonly found in today's EPUB's for instance, to have an idea whether this would be useful at all.

In any case, keeping it as is may be safer for a first version?

Reply:
> In any case, keeping it as is may be safer for a first version?

Yes